### PR TITLE
Providing a basic documentation

### LINKED ISSUE #12: Add documentation contents
The project currently does not come with a proper documentation. So we should take care starting to document the most important concepts about the simulation itself, about the differences between opportunistic resources and common resources as well as how to extend the simulation to adapt it to the required use case.

### LINKED ISSUE #52: Check usage of Sphinx-Click for automatic CLI documentation
There is a package [sphinx-click](https://github.com/click-contrib/sphinx-click) that allows the automatic generation of documentation for click annotated CLIs. We should check the usability of generated documentation to decide if we want to include this.

### LINKED ISSUE #54: Add input file/size information to documentation
@tfesenbecker finished the documentation for the input file format including input files and their sizes for jobs.

> Job Information from HTCondor
> =============================
> 
> With lapis jobs can be created directly from HTCondor outputs. Therefore ClassAds describing a jobs requested and used
> resources can be gathered via the `condor_history` command and saved to a csv or json file.
> The following ClassAds are necessary in order to describe a job sufficiently:
> 
> * requested resources:
>     RequestCpus, RequestWalltime, RequestMemory, RequestDisk
> * used resources:
>     RemoteWallClockTime, MemoryUsage, DiskUsage_RAW, RemoteSysCpu, RemoteUserCpu
> * additional job information:
>     QDate
> 
> If csv is chosen as input file format every line represents a job, columns should be seperated by spaces, commentaries should be marked by simple quotation marks.
> 
> If information about a jobs input files are passed to lapis a json file should contain job descriptions because this file format allows for nested structures.
> In this case the json file should contain an array of objects, each representing a job.
> 
> 
> ClassAds containing information about a jobs input files are not part of a jobs standard ClassAds in HTCondor but can
> be extracted via external tools (e.g. job submission tools) and stored as `Inputfiles` ClassAd.
> Alternatively this information can be added to the job input file manually.
> 
> 
> The `Inputfile` ClassAd contains dictionary with the input file names serving as keys and subdictionaries further describing the input files.
> These subdictionaries provide
> 
> * "filesize":
>     the files total size in MB
> 
> * "usedsize":
>     the amount of data the job actually reads from this file in MB
> 
> Example:
> 
> .. code-block:: JSON
> 
>         [
>             {
>                 "QDate": 1567169672,
>                 "RequestCpus": 1,
>                 "RequestWalltime": 60,
>                 "RequestMemory": 2000,
>                 "RequestDisk": 6000000,
>                 "RemoteWallClockTime": 100.0,
>                 "MemoryUsage": 2867,
>                 "DiskUsage_RAW": 41898,
>                 "RemoteSysCpu": 10.0,
>                 "RemoteUserCpu": 40.0,
>             },
>             {
>                 "QDate": 1567169672,
>                 "RequestCpus": 1,
>                 "RequestWalltime": 60,
>                 "RequestMemory": 2000,
>                 "RequestDisk": 6000000,
>                 "RemoteWallClockTime": 100.0,
>                 "MemoryUsage": 2867,
>                 "DiskUsage_RAW": 41898,
>                 "RemoteSysCpu": 10.0,
>                 "RemoteUserCpu": 40.0,
>                 "Inputfiles": {
>                     "a.root": {
>                         "filesize": 25000,
>                         "usedsize": 20000
>                     },
>                     "b.root": {
>                         "filesize": 25000,
>                         "usedsize": 20000
>                     }
>                 }
>             }
>         ]
>

_Originally posted by @tfesenbecker in https://github.com/MatterMiners/lapis/pull/51#issuecomment-549044163_

### PR DESCRIPTION
**Pull request by eileen-kuehn**
This pull request includes basic documentation on how to use lapis and the concepts behind.

* [x] Documentation for monitoring state and objects of simulation

The current state of the documentation can be previewed at https://lapis-sim.readthedocs.io/en/feature-documentation/.

Closes #12, closes #54, closes #52.

### CODE REVIEW

**Comment by maxfischer2781** [comment]
While incomplete, this is quite an improvement. Can we merge this early maybe, and add further changes into a new PR?

**Comment by eileen-kuehn** [comment]
@maxfischer2781, I will try to provide a reviewable version until end of the day. Thanks 😄

File: docs/source/glossary.rst [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+.. glossary::
+
+   Allocation
+      Information about the amount of resources being acquired by a :term:`drone`
+      without evaluating the effectiveness of use.
+
+      .. note::
+
+         The general concept of allocation and :term:`utilisation` is introduced
+         by COBalD to internally decide about which resources to integrate or
```
Comment (maxfischer2781) [inline docs/source/glossary.rst]:
```suggestion
         by COBalD to internally decide which resources to integrate or
```

File: docs/source/topics/concept.rst [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+* :term:`Drones <Drone>`, and
+* the :term:`Scheduler`,
+
+If you are planning to adapt the simulation for your specific use case, please
+consider the different components to determine what and where to extend functionality.
+
+Job Generator
+~~~~~~~~~~~~~
+
+The Job Generator works on the job input files. It takes care to
```
Comment (maxfischer2781) [inline docs/source/topics/concept.rst]:
```suggestion
The Job Generator processes any job input files. It takes care to
```

File: docs/source/topics/concept.rst [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+:term:`drones <Drone>` is adapted dynamically by the
+:term:`pool controller <Controller>` for dynamic pools.
+
+.. autoclass:: lapis.pool.Pool
+.. autoclass:: lapis.pool.StaticPool
+
+Controllers
+~~~~~~~~~~~
+
+Each :term:`pool` is started with an appropriate controller. Each controller runs
```
Comment (maxfischer2781) [inline docs/source/topics/concept.rst]:
```suggestion
Each :term:`pool` is managed by a parent controller. Each controller runs
```

File: docs/source/topics/concept.rst [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+
+.. warning::
+
+    Drones are not yet fully employed in LAPIS. They already run independently
+    but do not handle termination themselves.
+
+Scheduler
+~~~~~~~~~
+
+The scheduler is the connecting component between the :term:`jobs <Job>` in the
```
Comment (maxfischer2781) [inline docs/source/topics/concept.rst]: This section is very technically detailed. Perhaps a more general description matching other scheduler implementations would be more appropriate.

File: docs/source/topics/support.rst [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+requested resources:
+    RequestCpus, RequestWalltime, RequestMemory, RequestDisk
+
+used resources:
+    RemoteWallClockTime, MemoryUsage, DiskUsage_RAW, RemoteSysCpu, RemoteUserCpu
+
+additional job information:
+    QDate
+
+If csv is chosen as input file format every line represents a :term:`job`,
```
Comment (maxfischer2781) [inline docs/source/topics/support.rst]: Can you take another look at this paragraph and the 3 following ones? Some formulations read a bit jarring, e.g. "information about a jobs input files" has two plurals. Not sure if these are on purpose.

File: docs/source/topics/concept.rst (hunk 3 above)
Comment (eileen-kuehn) [reply docs/source/topics/concept.rst]: I wouldn't like to put the information about parent here... This is a piece of information that we do not cover. What do you think about:

```suggestion
Each :term:`pool` is managed by a controller. Each controller runs
```

File: docs/source/topics/concept.rst (hunk 4 above)
Comment (eileen-kuehn) [reply docs/source/topics/concept.rst]: What do you mean? Could you clarify a bit?
My intention is to follow the descriptions before. Those followed the actual implementation within LAPIS, so this is what I also did here for the scheduler.

File: docs/source/topics/support.rst (hunk 5 above)
Comment (eileen-kuehn) [reply docs/source/topics/support.rst]: Yes, this one is on purpose, meaning the "input files of a job". What do you propose to improve the reading?

Comment (eileen-kuehn) [reply docs/source/topics/support.rst]: Sorry, this was outdated information that was still part of @tfesenbecker's commit. I adapted the information to our current plans.

**Pull request merged**